Services/Github/useGetRepo.ts

```diff
@@ -0,0 +1,57 @@
+import { useState, useEffect, useCallback, useRef } from 'react';
+import type { Repo } from './types';
+
+export const useGetRepo = (repo: string) => {
+  const [data, setData] = useState<Repo>();
+  const [error, setError] = useState<string | null>(null);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const hasFetched = useRef(false);
+
+  const fetchRepo = useCallback(async () => {
+    try {
+      const repoResponse = await fetch(`https://api.github.com/repos/${repo}`);
+
+      let repoData: Repo;
+      if (repoResponse.ok) {
+        repoData = await repoResponse.json();
+
+        if (repoData && !repoData?.contributors) {
+          const contributorsResponse = await fetch(repoData.contributors_url);
+
+          if (contributorsResponse.ok) {
+            repoData.contributors = await contributorsResponse.json();
+          } else {
+            repoData.contributors = [];
+            console.error(`Failed to fetch contributors for repo: ${repo}`);
+          }
+        }
+
+        setData(repoData);
```

Comment:
Mutations are bad practice.

Here's an example to avoid it (mind any typos):

```suggestion
    let contributors = [];

      if (repoData && !repoData?.contributors && repoData.contributors_url) {
         try {
           const resContribs = await fetch(repoData.contributors_url);
           contributors = resContribs.ok
             ? resContribs.json()
             : [];
         } catch () {
           console.error(`Failed to fetch contributors for repo: ${repo}`);
         }
       }

       setData({
           ...repoData,
           contributors,
        });
```